subsys/zigbee/cli/zigbee_cli.c

```diff
@@ -0,0 +1,90 @@
+/*
+ * Copyright (c) 2020 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <stdio.h>
+#include <stdbool.h>
+#include <stddef.h>
+#include <init.h>
+#include <shell/shell.h>
+
+#include <zb_nrf_platform.h>
+#include "zigbee_cli.h"
+
+/* Endpoint used by the Zigbee CLI. */
+static zb_uint8_t cli_ep;
+
+/* Zigbee CLI debug mode indicator. */
+static zb_bool_t debug_mode = ZB_FALSE;
+
+LOG_MODULE_REGISTER(cli, CONFIG_ZIGBEE_SHELL_LOG_LEVEL);
+
+
+/**@brief Returns the Endpoint number used by the CLI.
+ */
+zb_uint8_t zb_cli_get_endpoint(void)
+{
+	return cli_ep;
+}
+
+/**@brief Sets the Endpoint number used by the CLI.
+ */
+zb_void_t zb_cli_set_endpoint(zb_uint8_t ep)
+{
+	cli_ep = ep;
+}
+
+/**@brief Configures CLI endpoint by setting number of endpoint to be used
+ * by the CLI and for chosen endpoint registers CLI endpoint handler.
+ */
+zb_void_t zb_cli_configure_endpoint(void)
+{
+	zb_zcl_globals_t *zcl_ctx = zb_zcl_get_ctx();
+	zb_af_endpoint_desc_t *cli_ep_desc;
+
+	/* Verify that ZCL device ctx is present at the device as for example,
+	 * network coordinator sample doesn't have one by default.
+	 */
+	if (zcl_ctx->device_ctx == NULL) {
+		LOG_ERR("Register ZCL ctx at Zigbee device to use CLI ZCL commands.");
+		return;
+	}
+
+	/* Set endpoint number to use from KConfig. */
+#ifdef CONFIG_ZIGBEE_SHELL_ENDPOINT
+	zb_cli_set_endpoint(CONFIG_ZIGBEE_SHELL_ENDPOINT);
```

Comment:
For the future please use `IS_ENABLED` macro. It will ensure that code is always compiled.